packages/api/src/external/carequality/__tests__/cq.demographics.test.ts

```diff
@@ -0,0 +1,72 @@
+import { PatientResource } from "@metriport/ihe-gateway-sdk";
+import { patientResourceToNormalizedLinkDemographics } from "../patient-demographics";
+
+describe("CQ demographics", () => {
+  const patientResource: PatientResource = {
+    birthDate: " 1900-01-01 ",
+    gender: "female",
+    name: [
+      {
+        family: "Smith",
+        given: ["Katherine", "Katy "],
+      },
+    ],
+    address: [
+      {
+        line: ["400 Awesome Rd."],
+        city: "San Francisco",
+        state: "CA",
+        postalCode: "99999-4040",
+      },
+      {
+        line: ["401 Awesome Rd.", "Apt 1b"],
+        city: "San Francisco ",
+        state: "CA",
+        postalCode: "99999",
+        country: "USA",
+      },
+    ],
+    telecom: [
+      {
+        system: "phone",
+        value: "(888)8887777",
+      },
+      {
+        system: "email",
+        value: " katy2020@GMAIL.COM",
+      },
+    ],
+    identifier: [],
+  };
+  it("normalization", async () => {
+    const linkDemographics = patientResourceToNormalizedLinkDemographics(patientResource);
+    expect(linkDemographics).toMatchObject({
+      dob: "1900-01-01",
+      gender: "female",
+      names: [
+        { firstName: "katherine", lastName: "smith" },
+        { firstName: "katy", lastName: "smith" },
+      ].map(name => JSON.stringify(name, Object.keys(name).sort())),
```

Comment:
To make sure we always have the same order in the output when matching:

`{ firstName... lastName...}' will not match '{ lastName ... firstName...}' if they're strings.

If we're worried about performance, I can essentially create our own function for this (i.e. `${name.firstName} ${name.lastName}`) but I figured JSON.stringify does the job anyway.

Reply:
Gotcha.

Not concerned about performance there, mostly trying to avoid writing unnecessary code - and this being identified as a new pattern that folks start to follow even when not needed.

Def not a thing for this PR, but to consider when doing this again: maybe have a function to do it, so we can use it like `.map(stringifySorted)`